### ISSUE TITLE
QTVCP Deprecation Warnings

### ISSUE THREAD

**Issue by satiowadahc** [LinuxCNC/linuxcnc#979]
Here are the steps I follow to reproduce the issue:

 1. Using our own custom UI, based off qtvcp with some pyqtvcp
 2. Running CI testing, which tests 90% of our functionality
 3. Currently off Rene-Devs branch of limuxcmc...  Will try again momentarily off master

## This is what I expected to happen:
```bash
../../../../usr/lib/python3/dist-packages/qtvcp/core.py:118
  /usr/lib/python3/dist-packages/qtvcp/core.py:118: PyGIDeprecationWarning: GObject.SIGNAL_RUN_FIRST is deprecated; use GObject.SignalFlags.RUN_FIRST instead
    'toolfile-stale': (GObject.SIGNAL_RUN_FIRST, GObject.TYPE_NONE, (GObject.TYPE_PYOBJECT,)),

../../../../usr/lib/python3/dist-packages/qtvcp/core.py:142
  /usr/lib/python3/dist-packages/qtvcp/core.py:142: PyGIDeprecationWarning: Since version 3.11, calling threads_init is no longer needed. See: https://wiki.gnome.org/PyGObject/Threading
    GObject.threads_init()

../../../../usr/lib/python3/dist-packages/qtvcp/core.py:143
  /usr/lib/python3/dist-packages/qtvcp/core.py:143: PyGIDeprecationWarning: GObject.timeout_add is deprecated; use GLib.timeout_add instead
    GObject.timeout_add(100, self.update)

/usr/lib/python3/dist-packages/qtvcp/widgets/origin_offsetview.py:191: DeprecationWarning: This method will be removed in a future version of Python. Use 'locale.format_string()' instead.
    self.tabledata[row][column] = locale.format(tmpl, i[column])

/usr/lib/python3/dist-packages/qtvcp/widgets/origin_offsetview.py:187: DeprecationWarning: This method will be removed in a future version of Python. Use 'locale.format_string()' instead.
    self.tabledata[row][column] = locale.format(degree_tmpl, rot)

 /usr/lib/python3/dist-packages/qtvcp/qt_action.py:209: DeprecationWarning: PY_SSIZE_T_CLEAN will be required for '#' formats
    self.cmd.program_open(str(fname))


```

## Information about my hardware and software:

 * I am using this Linux distribution and version (often, shown by `lsb_release -a`):
 Ubuntu 20.04, Replicable on Linux Mint and Lubuntu
 * I am using this kernel version (shown by `uname -a`): 
Linux CHAD-I7-LINUX 5.4.61-rt37 #1 SMP PREEMPT_RT Wed Sep 2 11:08:45 CST 2020 x86_64 x86_64 x86_64 GNU/Linux

 * I am running ...
   * [ ] A binary I built myself from rene-dev/limuxcmc
 * I am using this LinuxCNC version: 2.9.0
 * I am using this user interface (GUI): Custom QTVCP
 * I am using this interface hardware vendor and chipset: Sim machine

**Comment by satiowadahc** [comment]
I've pulled in master and it seems to have cleaned these up. I might dedicate some time to documentation and code testing today

**Comment by satiowadahc** [comment]
This was closed by #1195
